### PR TITLE
Events API Reference swagger and manifest [REVIEW - DO NOT MERGE]

### PR DESCRIPTION
**Pull request by djwendo**

### CODE REVIEW

**Comment by djwendo** [comment]
@francoisledroff now that the API only contains specific endpoints, what should the proper swagger title be, and should we be adding different/more tags in order to separate the endpoints into groups that would be useful to the end-user?

**Comment by francoisledroff** [comment]
Yes I believe it would be easier to read through the doc with a bit more ordering and a few tags.
here is How I would advise to list them:

A first section/tag grouping the endpoints that are about Creating updating Custom event providers and associated metadata 

* POST /events/{consumerOrgId}/{projectId}/{workspaceId}/providers Creates an Adobe I/O Events Provider
* PUT /events/{consumerOrgId}/{projectId}/{workspaceId}/providers/{providerId} Update an Adobe I/O Events Provider
* PATCH /events/{consumerOrgId}/{projectId}/{workspaceId}/providers/{providerId} PATCH an Adobe I/O Events Provider using json-patch
* DELETE /events/{consumerOrgId}/{projectId}/{workspaceId}/providers/{providerId} Delete Adobe I/O Events Provider by Id


* POST /events/{consumerOrgId}/{projectId}/{workspaceId}/providers/{providerId}/eventmetadata Creates an Adobe I/O Events Provider Event Metadata
* PUT /events/{consumerOrgId}/{projectId}/{workspaceId}/providers/{providerId}/eventmetadata/{eventCode} Update an Adobe I/O Events Provider Event Metadata
* PATCH /events/{consumerOrgId}/{projectId}/{workspaceId}/providers/{providerId}/eventmetadata/{eventCode} PATCH an Adobe I/O Events Provider Event Metadata
* DELETE /events/{consumerOrgId}/{projectId}/{workspaceId}/providers/{providerId}/eventmetadata Delete All the Event Metadata associated with the given Adobe I/O Events Provider providerId
* DELETE /events/{consumerOrgId}/{projectId}/{workspaceId}/providers/{providerId}/eventmetadata/{eventCode} Delete the Event Metadata associated with the given Adobe I/O Events Provider providerId and eventCode

A second section grouping the endpoints that are about reading Providers and associated metadata 

* GET /events/{consumerOrgId}/providers Get All Adobe I/O Events Providers entitled to the provided consumerOrgId
* GET /events/providers/{id} Get Adobe I/O Events Provider by Id
* GET /events/providers/{providerId}/eventmetadata/{eventCode} Get the Event Metadata associated with a given Adobe I/O Events Provider providerId and eventCode
* GET /events/providers/{providerId}/eventmetadata List the Event Metadata associated with a given Adobe I/O Events Provider providerId
* GET /events/providers/{providerId}/eventmetadata/{eventCode}/sample_event Get a sample event HAL json model of the provided Adobe I/O Events Provider id and event code
* HEAD /events/providers/{providerId}/eventmetadata/{eventCode}/sample_event Check if a sample event is available for the provided Adobe I/O Events Provider id and event code

a 3rd section around registrations:

* POST /events/organizations/{consumerOrgId}/integrations/{integrationId}/registrations Creates a Client Application's Webhook / Journal registration
* GET /events/organizations/{consumerOrgId}/integrations/{integrationId}/registrations Get all registration details
* GET /events/organizations/{consumerOrgId}/integrations/{integrationId}/registrations/{registrationId} Get a specific registration details
* DELETE /events/organizations/{consumerOrgId}/integrations/{integrationId}/registrations/{registrationId} Delete a specific registration details

a 4th about tracing:

* GET /events/organizations/{consumerOrgId}/integrations/{integrationId}/tracing/{registrationId} Get a specific WebHook tracing details

and finally about journaling:

* GET /events/organizations/{consumerOrgId}/integrations/{integrationId}/{registrationId}/validate Validate and get all Events from journal 
* GET /events/organizations/{consumerOrgId}/integrations/{integrationId}/{registrationId} Get all Events from journal

**Comment by djwendo** [comment]
@francoisledroff - Is the title for the API (I/O Management API) still appropriate, or should this be specific to Events API? Custom Events API?

File: events-api-reference.yaml [hunk 1]
```diff
@@ -0,0 +1,1070 @@
+swagger: '2.0'
+info:
+  title: I/O Management API
```
Comment (francoisledroff) [inline events-api-reference.yaml:3]: `Adobe I/O Events API` would be more appropriate I guess

File: events-api-reference.yaml [hunk 2]
```diff
@@ -0,0 +1,1070 @@
...
+  version: '2.0'
+host: api-stage.adobe.io
```
Comment (francoisledroff) [inline events-api-reference.yaml:5]: so this is for stage ? is that on purpose ?

**Comment by francoisledroff** [comment]
Hey Wendy, is there an associated swagger docs UI somewhere ? that would ease its review.

File: events-api-reference.yaml (hunk 1 above)
Comment (djwendo) [reply events-api-reference.yaml]: Updated to Adobe I/O Events API

File: events-api-reference.yaml (hunk 2 above)
Comment (djwendo) [reply events-api-reference.yaml:5]: This came from the file that was provided: https://git.corp.adobe.com/pages/adobe-apis/api-mgmt-docs/#/events

The base is shown at the link above as `api-stage.adobe.io` and then when you click the link in the swagger example to view the json, that is the host url.

**Comment by djwendo** [comment]
@francoisledroff to view the swagger more easily, you can copy the raw yaml from here: 

https://raw.githubusercontent.com/AdobeDocs/adobeio-events/cf8d1a4edddf24f6fc4b88eeaae41b08cdd12d5d/events-api-reference.yaml

and paste it into editor.swagger.io. 

You will see that the endpoints are in the order you suggested. The 'tags' element at the top of the yaml sets the order in which they will render, and then applying a tag to each of the 'paths' is what groups them together. So I didn't change the 'paths', I just applied the appropriate 'tags'.

File: events-api-reference.yaml (hunk 2 above)
Comment (francoisledroff) [reply events-api-reference.yaml:5]: yes prod is also available at https://git.corp.adobe.com/pages/adobe-apis/api-mgmt-docs/#/ see the `select a spec`drop down 
the associated file is there : https://git.corp.adobe.com/adobe-apis/api-mgmt-docs/blob/master/v2/api-docs-prod.json

**Comment by francoisledroff** [comment]
> editor.swagger.io.

thanks ! it's looking good !

**Pull request merged**